### ISSUE TITLE
error in NIFTItoNRRD handling of space directions

### ISSUE THREAD

**Issue by styner** [BRAINSia/BRAINSTools#308]
DWIConvert incorrectly convert space directions from NIFTI to NRRD in that it seems to simply write axis-aligned space directions. 
Take any NRRD DWI with none-axis-aligned space directions convert it to NIFTI and then convert it back to NRRD and it will have lost the correct space directions information

**Comment by hjmjohnson** [comment]
Thank you.  The NIFTI implementation has been problematic for us.  It was contributed code that we did not vet well enough.  

I am working on DWIConvert and NIFTI support this month.  Perhaps before December 1st it will be fixed.

**Comment by styner** [comment]
great, thanks. I am using a workaround (hacking the nrrd header) in the mean time.

**Comment by hjmjohnson** [comment]
@styner Martin,
A command line use case would be really helpful.  What command line flags are you using to convert to/from nifti.

Additionally, I am concerned that the nifti bvec and bval files are not in the correct format.  Other tools that convert dicom to nifti have a different layout for the bvec and bval files.

Hans

**Comment by styner** [comment]
Yeah, but I used DWIConvert to get to the NIFTI, i.e. 
DWIConvert --conversionMode NrrdToFSL --inputVolume $NrrdFile -o $targetNII --outputBValues $targetNII:r:r_bval.txt --outputBVectors $targetNII:r:r_bvec.txt 
DWIConvert --conversionMode FSLToNrrd --inputBVectors $targetNII:r:r_bvec.txt --inputBValues  $targetNII:r:r_bval.txt   --fslNIFTIFile $targetNII --outputVolume $targetNrrd

using a NRRD file with non-axis aligned space directions is all you need to see this bug

**Comment by hjmjohnson** [comment]
Thank you.  I’ll look into it ASAP in the next week or two.

 

Hans

**Comment by hjmjohnson** [comment]
@styner Martin,   I think this is now fixed in #315 

I have a test suite of 85 visualizations using both fslview and Slicer to ensure that corrections are done.  I'm through 15 visualization validations (a problematic phillips, a GE, and a Siemens), and will complete the rest of the data visualizations tomorrow.

dcm2niix    -> view in fslview
DWIConvert:DicomToNrrd -> View with Slicer
DWIConvert:DicomToFSL -> view in fslview
dcm2niix, DWIConvert:FSLToNrrd -> View in Slicer
DWIConvert:DicomToFSL, DWIConvert:FSLToNrrd ->View in Slicer

[VisualizationValidation.pdf](https://github.com/BRAINSia/BRAINSTools/files/644847/VisualizationValidation.pdf)

I also added a new feature described in #314 , and updated the version number to 4.8.0.


PS:  I reported a bug about dtiestim.  I think it is producing bad results.   It look me a while to find a substitute tool that was compatible with Slicer to do the visualizations.
